### PR TITLE
Fix bug in multiindex series groupby where sort argument is ignored

### LINKED ISSUE #9444: sort=False ignored in Series groupby on MultiIndex levels
Example:

    import pandas
    def geta():
        i = pandas.MultiIndex.from_tuples([(1, 2, 'a', 0),
                                  (1, 2, 'a', 1),
                                  (1, 1, 'b', 0),
                                  (1, 1, 'b', 1),
                                  (2, 1, 'b', 0),
                                  (2, 1, 'b', 1)], names=['a', 'b', 'c', 'd'])
        a = pandas.Series([0, 1, 2, 3, 4, 5], index=i)
        return(a)

    for dosort in [True, False]:
        a = geta()
        b = a.groupby(level=['a', 'b'], sort=dosort).first()
        a = None
        print('%s, sort=%s, \n%s' % (geta.__name__, dosort, b))

Output:

    geta, sort=True,
    a  b
    1  1    2
       2    0
    2  1    4
    dtype: int64
    geta, sort=False,
    a  b
    1  1    2
       2    0
    2  1    4
    dtype: int64

Also see: https://github.com/pydata/pandas/pull/9076
Possibly related to: https://github.com/pydata/pandas/issues/8868

### PR DESCRIPTION
**Pull request by cottrell**
PR as per. 

One test added. Test fails without the fix.

closes https://github.com/pydata/pandas/issues/9444

### CODE REVIEW

**Comment by jreback** [comment]
looks good
ping when green

File: pandas/tests/test_groupby.py [hunk 1]
```diff
@@ -107,6 +107,15 @@ def setUp(self):
                                       'E': np.random.randn(11),
                                       'F': np.random.randn(11)})
 
+        index = MultiIndex(levels=[[1, 2], [1, 2]],
+                           labels=[[0, 0, 0, 0, 1, 1], [1, 1, 0, 0, 0, 0]],
+                           names=['a', 'b'])
+        self.mseries = Series([0, 1, 2, 3, 4, 5], index=index)
+        index = MultiIndex(levels=[[1, 2], [1, 2]],
+                           labels=[[0, 0, 1], [1, 0, 0]],
```
Comment (jreback) [inline pandas/tests/test_groupby.py]: actually
move these into the actual test function
they only need to be created once
if they are in the setup they will be for every test - which is not necessary

**Comment by jreback** [comment]
@cottrell looks good now. thanks. ping when green.

**Comment by jreback** [comment]
does this also fix #8868  ?

**Comment by cottrell** [comment]
@jreback It does not seem to in my test of #8868. Probably a similar issue but elsewhere. I can probably have a look later this week.

**Comment by jreback** [comment]
ok gr8 thanks

**Comment by jreback** [comment]
thanks!

**Pull request merged**